### ISSUE TITLE
Gathering: MountFly Mode fails if Flying Mount Takeoff Mode is set to 'Manual'

### ISSUE THREAD

**Issue by Nox13last** [awgil/ffxiv_visland#18]
The issue is pretty self-explanatory. If the player's game isn't set to Flying Mount Takeoff Mode: **Automatic**, the player (and their mount) hops around in circles, unable to reach the airborne waypoint.

For now, it seems the solution is to just open the Character Configuration and flip that switch before executing this route (and turning it back to Manual, when done, if the player likes to be able to jump without entering flight).

From an outsider's perspective, maybe have the automator double-tap jump before ascending (this would only affect Manual Mode; Auto Mode probably wouldn't see anything unusual).

**Comment by Nox13last** [comment]
I'm going to assume; with the passage of other tickets being created and answered and closed, and this one being stale by almost two months; that this is a #wontfix ticket.

**Comment by awgil** [comment]
I plan to do a big pass on revising whole movement stuff soon-ish.

**Comment by Nox13last** [comment]
That's cool. With the small exception of this one need for a workaround, this plugin has made gathering stuff for extended periods of time bearable.

Don't suppose you consider expanding this concept beyond the Island? Getting the gubbins together for the Diadochos set is its own form of misery.

**Comment by Jaksuhn** [comment]
It should mostly work outside the island if you have something like pandora gathering. Anyway I fixed the problems for manual flyers.

**Comment by Nox13last** [comment]
> It should mostly work outside the island if you have something like pandora gathering. Anyway I fixed the problems for manual flyers.

I apologise. I would like to reopen this ticket. Feel free to close it again if it is too far outside of scope and you lot don't feel it warrants pursuit, but I have tried this principle, and it is not successful. **Almost.** But not.
Included in the post attachments is a gathering track for [Dark-Chestnut-Logs](https://github.com/awgil/ffxiv_visland/files/13453835/Dark-Chestnut-Log.json)

It follows this route through the Dravanian Forelands.
![image](https://github.com/awgil/ffxiv_visland/assets/1292492/17212a06-7bfa-4bed-8328-e1699844b78a)
(The augmented lines in this image and following media are provided in-engine by NightmareXIV's Splatoon (in the [PunishXIV repo](https://love.puni.sh/ment.json)), and are set (for the purposes of investigation) to tell me if a node has spawned or is unavailable/depleted, and in general, to tell me how I should fly.)
Here is a [layout file](https://github.com/awgil/ffxiv_visland/files/13453845/Dark-Chestnut-Log-Splatoon.json) if you'd like to see that for yourself.

There is an issue. Upon reaching the waypoint, if the player is mounted, Visland _immediately_ attempts to saddle-up again. This spams a bunch of "Can't do that while jumping" (if the mount is tall and the rider has to fall to the ground) and "You can't do that while casting" errors. The latter happens because Pandora is attempting to dismount and start a gathering operation, but Visland has already lost interest and is attempting to leave.
Here is a video demonstrating this. Sorry for the poor quality - I had to churn this through 8mb.video to stop GitHub from choking and suffocating on it.
https://github.com/awgil/ffxiv_visland/assets/1292492/042548c5-4ae6-4de0-848a-f4db311165e3

I've had a good think about it. I know this isn't really what Visland is for, but having to gather 1800 gubbins, I'll take what I can get. I notice that Visland supports Waits and Conditions (as seen in /xldata)...
![image](https://github.com/awgil/ffxiv_visland/assets/1292492/1ac011fb-5d18-4574-a047-97bd49f079d0)
...but I don't think it's possible to set this up in such a way as "Wait until player has finished gathering, unless the node didn't exist in the first place." Seems like Visland would either miss a node that exists, or wait forever to finish a node that can't start because it never became active. But it needs to wait for _something_ so it doesn't lose interest and depart from an extant node.
It works if Visland never saddles-up (as being on foot gives Pandora enough time to snag the node, and the player isn't casting or falling), but terrain means a lot of convoluted foot-pathing to clear something a mount could clear trivially.
![image](https://github.com/awgil/ffxiv_visland/assets/1292492/f126a7ae-ae99-4790-8283-b706a3aff424)

Any ideas, or #wontfix?

**Comment by Jaksuhn** [comment]
Can you not just put a second or two wait at each waypoint to account for the dismounting time?

**Comment by Nox13last** [comment]
> Can you not just put a second or two wait at each waypoint to account for the dismounting time?

I tried that. It didn't work. Even if I set the Wait parameter to something conspicuously high, like 30 seconds, the player would just embark immediately anyway.
I think the Wait parameter goes hand-in-hand with the Condition parameter, and it goes entirely ignored if the Condition is not set.
I could be mistaken (I kind of burned myself out thinking about it), but there didn't seem to be a way to just get Visland to **wait for the sake of waiting** (as opposed to **waiting until a finite condition is met**).

I noticed also that Visland has support for "do an action". The list of possible actions it can take includes literally every class action, PVP action, Variant action, and Role-Playing setpiece action (like _[Noun] of the Seventh Dawn_), but does not support General Actions (for instance: _/mount_, _/dismount_, _/ga Jump_, _/ga Sprint_). I cannot fathom why this would or would not be the case.

By my reckoning, there are three idealised solutions:
- Have there be an unconditional Wait.
- Have the _Normal_ movement mode cause a dismount (so the player can approach a node on foot). _Normal_ mode does not change the player's current mode of travel.
- Have support for "parse this text command upon reaching a waypoint", not just an auto-populated dropdown.

As I see it, there are a bunch of _half-_solutions, but they all come up short for one reason or another. ⵌ3 on its own would be absolutely sick.
